File: 5/max_depth_nary_tree.go
package main

import "fmt"

type Node struct {
	val      int
	children []Node
}

func maxDepth(root *Node) int {
	if root == nil {
		return 0
	}
	if root.children == nil {
		return 1
	}
	res := 1
	children := root.children
	for len(children) > 0 {
		var newChildren []Node
		for _, node := range children {
			newChildren = append(newChildren, node.children...)
		}
		children = newChildren
		res++
	}
	return res
}

func main() {
	root := Node{1, []Node{
		Node{2, []Node{
			Node{3, nil},
			Node{4, nil},
		}},
		Node{5, nil},
	}}
	fmt.Println(maxDepth(&root))
}

File: 5/max_depth_nary_tree.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val, children):
        self.val = val
        self.children = children
"""
class Solution:
    def maxDepth(self, root: 'Node') -> int:
        if not root:
            return 0
        if root.children == None or root.children == []:
            return 1
        res = 0
        children = [root]
        while children != []:
            new_children = []
            for n in children:
                new_children += n.children
            children = new_children
            res+=1
        return res
